packages/apollo-codegen-swift/src/helpers.ts

```diff
@@ -132,6 +135,24 @@ export class Helpers {
     );
   }
 
+  /**
+   * Returns the internal parameter name for a given property name.
+   *
+   * If the property name is valid to use, it's returned directly. Otherwise it's prefixed with an
+   * underscore and modified until it's unique among the given property set.
+   * @param propertyName The name of the property.
+   * @param properties A list of properties that should be consulted when producing a unique name.
+   * @returns The name to use for the internal parameter name for the property.
+   */
+  internalParameterName(
+    propertyName: string,
+    properties: { propertyName: string }[]
+  ): string {
+    return SwiftSource.isValidParameterName(propertyName)
+      ? propertyName
+      : makeUniqueName(`_${propertyName}`, properties);
```

Comment:
So am I correct in understanding that `makeUniqueName` is necessary in case there are properties `self` and `_self` on something, since just appending an underscore would cause a conflict?

Reply:
In case there’s a property `_self`. This code only runs if there’s a property named `self` since that’s what we’re trying to modify (`self` isn’t valid by itself, so we turn it into `_self`, and then `makeUniqueName` ensures we don’t collide with an existing property `_self`).

I could also have done `makeUniqueName(`${propertyName}_`, properties)` instead but I prefer the aesthetic of using `_self` instead of `self_` in the common case. I admit that `_self_` looks weird, but that should be vanishingly rare unless someone is being silly.

That said, if you’d prefer to have it start as `self_` I won’t object.

Reply:
Was mostly trying to double-check why this had to exist - it'd need to exist even if we went with the `self_` method, though. I think using the underscore-as-prefix works better with *OS conventions, so nbd.